comanage_utils.py

```diff
@@ -81,20 +82,27 @@ def call_api2(method, target, endpoint, authstr, **kw):
 
 def call_api3(method, target, data, endpoint, authstr, **kw):
     req = mkrequest(method, target, data, endpoint, authstr, **kw)
-    trying = True
-    currentTimeout = MIN_TIMEOUT
-    while trying:
+    retries = 0
+    currentTimeout = TIMEOUT_MIN
+    requestingStart = time.time()
+    payload = None
+    while payload == None:
         try:
             resp = urllib.request.urlopen(req, timeout=currentTimeout)
+            if retries > 0:
+                print(f"Succeeded for request {req.full_url} after {retries} retries.")
             payload = resp.read()
-            trying = False
         except urllib.error.URLError as exception:
-            if currentTimeout < MAX_TIMEOUT:
-                currentTimeout *= TIMEOUTMULTIPLE
+            if retries < MAX_RETRIES:
+                print(f"Error: {exception} for request {req.full_url}, sleeping for {currentTimeout} seconds and retrying.")
+                time.sleep(currentTimeout)
+                currentTimeout *= TIMEOUT_MULTIPLE
+                retries += 1
```

Comment:
```suggestion
                retries += 1
                currentTimeout **= retries
```
Then you can get rid of `TIMEOUT_MULTIPLE`

Reply:
If I'm understanding correctly, wouldn't this lead to cool-down / timeout times of 5 seconds, 25 seconds, and then 15625 seconds (25 seconds ^ 3 third retry, or 4.34 hours)?

Reply:
Oh, you're right! You can still simplify with

```
retries +=1
current_timeout *= TIMEOUT_BASE
```